### ISSUE TITLE
Introtext and readmore aren't working in event details

### ISSUE THREAD

**Issue by jojo12** [jemproject/JEM-Project#1706]
in Joomla 5 JEM 4.2 when I use the general settings for event details (page), only introtext and readmore are not working
(Until now I never used this (and normally don't need) but there was such question in the forum)
I am in responsive style!

**Comment by mckillo** [comment]
Those options aren't active in the code:
block-line 189 components\com_jem\views\event\tmpl\default.php
block-line 195 components\com_jem\views\event\tmpl\responsive\default.php

It could be active but there is the review code, because in this case, the event only display the introtext if the user isn't logon and the read mode link is a link to register.

If you want those 2 buttoms in the JEM setting could be disable for the moment or we can think if active this feature in the event.

**Comment by mckillo** [comment]
What do we do with this?

**Comment by jojo12** [comment]
for me, we can let it. But this was a post in forum!

**Comment by mckillo** [comment]
As I said, this requires developing the code, I suppose that was the intention at the time but it was left undeveloped.
It can be left open to decide in the future. Now, this issue is the very low priority.

**Comment by Heklaterriol** [comment]
I have an other suggestion:
The readmore text does not make sense in the event detail view. Like in Joomla articles, where the readmore features is made for the category/blog view to read the full article.

My suggestion: To the modules with description text displayed, I would like to add the option "Show only Intro Text" right below the option "Description Length". This would be nice to display the whole Intro text, but not more.  
With "Description Length" only I have the options of an ugly cut in the middle of a text or alternatively the whole text.

If you agree to this I would try to implement it.

**Comment by jojo12** [comment]
(Please read first and hold in mind the special Help>Responsive>Teaser Module!

When I use the "readmore button" in eventdescription, this works fine in JEM-Teaser, even in JEM-Banner: there is a link with the text "readmore".

I agree, the cutting text with "Description Length" is not nice. But it's a general possibility to control the smallness of the modules of all creators (it's not nice, when they are too long.) Like that you don't need to put always a "readmore" into the eventdescription.

But it would be better, if there would be a link with text "Readmore". 
(I know, you can click onto the event title, to get the event detail view, but when there is a description which is suddenly cut, I think this is an error!

**Comment by mckillo** [comment]
I prefer that the 'Read more' function works as well. This way, the event editor can choose where to cut the intro text.
I'm working to fix it.

**Comment by mckillo** [comment]
"Read more..." works in the event list.

![Image](https://github.com/user-attachments/assets/ca26b017-bf86-4cc5-adc3-4b90d966f8da)

**Comment by mckillo** [comment]
Please test it, and if it works well, I can export it to other views and modules.

**Comment by jojo12** [comment]
I can't reproduce "without title" ? And does this make any sense in simplelist?
With title it's nice ->this is the modern look. 
Should this be in the tablelike look too? I don't think so --> what do you mean?

**Comment by jojo12** [comment]
An other question for the general JEM settings: This is a setting for tables (only simplelist or all tables? )
Should this be better in the Tab of List Layout?

**Comment by mckillo** [comment]
You can see the eventlist without title, set NO in:

![Image](https://github.com/user-attachments/assets/e435c915-1a8f-4b08-82e5-e1417630bc68)

Imagine you create a new menu for an event list view. In the options tab, you specify that this event list is for a specific category. All events in this category share the same title, so you want to use the date as the reference point for distinguishing them.

This commit is applied in responsive and legacy layout of this view.

**Comment by mckillo** [comment]
> An other question for the general JEM settings: This is a setting for tables (only simplelist or all tables? )
> Should this be better in the Tab of List Layout? (now this setting is in event page)

This new field is a setting menu of view. I think so, it's more flexible to setup in several call to the eventlist view.

**Comment by jojo12** [comment]
If I see right, then this menu setting "Show intro text of events" has nothing to do with general JEM setting>event page> "Show introtext". Correct?

**Comment by mckillo** [comment]
Yes, the new is a menu option of eventlist view. In General setting is a option of event view.

**Comment by jojo12** [comment]
With "Show intro text of events"=YES  simplelist shows always "readmore" even when there is no eventdescription. This is not nice in simplelist (too much). 
Normally  you should click the event title (or event date).
--> This should be changed!

**Comment by mckillo** [comment]
> With "Show intro text of events"=YES simplelist shows always "readmore" even when there is no eventdescription. This is not nice in simplelist (too much). Normally you should click the event title (or event date). --> This should be changed!

I agree. The event list view only displays 'Read more' if there is full text available. See last commit.

**Comment by mckillo** [comment]
> When title column =no then in the tablelike look (default_eventslist_small.php) the date has an other color than the links. (In my case all links are red, but the date is like normal text --> This should be changed!

If you want to link the date, have to change the display setting to:

![Image](https://github.com/user-attachments/assets/3aecad04-d0a0-4d93-b572-77f97ea6fcb1)

**Comment by mckillo** [comment]
Check in your database --> showtitle = 0 and showdetails = 1 value into xxxxx_jem_config table, if they are right then you must execute the content of else if in line 226 of \components\com_jem\common\views\tmpl\responsive\default_jem_eventslist.php 

![Image](https://github.com/user-attachments/assets/9e097730-7ee4-4f93-97ce-b55814af9c32)

And your date must have a link into <h4> tag

**Comment by jojo12** [comment]
Hi, showtitle=0 and showdetails=1 in jem_config_table.
But we are talking from different things:
I am talking from  **responsive\default_jem_eventslist_small.php**, you are talking from default_jem_eventslist.php !!!
(mine is responsive tablelike look and yours is responsive modern look!)

**Comment by mckillo** [comment]
Sorry, I have not change this table view. Forget it, I don't use it. Give me a little time to update it.

**Comment by mckillo** [comment]
Here you are the event list (small) without title and date as tittle and link.

![Image](https://github.com/user-attachments/assets/0a46f301-b23c-4e90-8bea-8da0877dbdac)

**Comment by jojo12** [comment]
BUG: no that was too quick: you have a mix between eventimage and eventtitle!

**Comment by jojo12** [comment]
it should be about like that:

![Image](https://github.com/user-attachments/assets/809eb677-37d5-4f17-8fb3-01d69f32d798)

**Comment by mckillo** [comment]
With last commit, works?

**Comment by jojo12** [comment]
No, not yet: in the body part lacks the image and the intro doesn't work

**Comment by jojo12** [comment]
in Legacy _**template without title**_ is added a column "Description", but it seems you can't set a width of this column, so it's very small and it's difficult to read.

**Comment by jojo12** [comment]
ALL in all I would say, very good! Thanks a lot!

**Comment by jojo12** [comment]
I tested more and found the following BUG:
Now it rights always the eventdescription even when there is no "readmore" button used.
[tested with RESPONSIVE template and in both TABLELIKE STYLE and MODERN STYLE]

**Comment by mckillo** [comment]
Clarification on Event Descriptions and Associated Attributes

**1. Storage of Event Descriptions:**

Historically, the full description of events was saved in the introtext field, while the fulltext field remained unused.
Users did not utilize the 'Read more' feature because it was non-functional.
After implementing the recent fix, if an event includes a 'Read more' tag (<hr id="system-readmore"> or <hr id='system-readmore'>), the introtext and fulltext are saved correctly.
If the 'Read more' tag is absent, the entire description is saved into introtext (db).

**2. Attributes:**

There are three attributes related to event descriptions:

_show_introtext_events_: A new field in the option menu of Eventlist view.
Determines whether to display the introtext in the event list view.
If an event contains fulltext, a 'Read more' link is added at the end of the introtext, linking to the event view.

_event_show_readmore_: Intended to control the display of the 'Read more' link when the description includes fulltext.
Currently, this code in JEM is inactive (if(0)), rendering it non-functional.

_event_show_intro_: Defined in the XML file but not utilized in the JEM code.
When set to 0, it should display only the fulltext. Introtext is shown in the previuos view (as eventlist view).
When set to 1, it should display both the introtext + fulltext as the event description in the event view.

**3. Configuration:**

If you prefer not to display descriptions in the event list, you need to set show_introtext_events to 0 in the options menu of the event list view.

**4. Next Steps:**

I will begin reviewing the code to address the functionality of the three attributes, focusing primarily on the last two. 
This issue isn't resolved yet; I need some time. 
However, today I'll be busy with family commitments. I'll attempt to work on it tonight or tomorrow.

**Comment by mckillo** [comment]
"EDIT: I think there's no definition of "descriptionwidth" column is not defined in the colgroup part. And when title is off, then there's no titlewidth"

it's true. Can we a new descriptionwitdh or can use the titlewidth?

**Comment by mckillo** [comment]
Point 2 was applied in the last commit. Now the administrators have control over how this event content is displayed:

- In the Event View: Administrators can choose to display either the combined "introtext" and "fulltext" (the entire article) or only the "fulltext.". To show only fulltext, set 'Show Introtext = Hide' in Settings - Event page - Event

![Image](https://github.com/user-attachments/assets/63ae5a42-4386-4f98-887b-c2807e3aa14c)

- In the EventList view: Administrators can opt to show only the event without or with a "Read More" link, providing a brief overview (introtext is set in the menu option). If  the admin don't want to show link to the end of introtext, set the "read More" link  to Hide in 
in Settings - Event page - Event

![Image](https://github.com/user-attachments/assets/614d11ae-3b5f-4b42-bc10-cfcccb023c89)

These settings allow for flexibility in presenting content in both views. Enjoy it.

**Comment by jojo12** [comment]
my eventdescription is without readmore

When I set menu>simplelist show Introtext in simplelist=YES, I get:


I am in RESPONSIVE MODERN STYLE, I get: " 0 Call to undefined method stdClass::get()" 


in RESPONSIVE TABLELIKE STYLE I get the whole eventdescription

**Comment by jojo12** [comment]
For better understanding:
Can we find a better wording for "fulltext", which is not the whole text?

**Comment by jojo12** [comment]
A proposal (I don't know, if this is possible)
When the "readmore" is used, then when without "introtext", it would be better to add a line like "continued" before the rest of the eventdescription.

**Comment by jojo12** [comment]
works fine

**Comment by mckillo** [comment]
> A proposal (I don't know, if this is possible) When the "readmore" is used, then when without "introtext", it would be better to add a line like "continued" before the rest of the eventdescription.

But I see it fine; a new article without intro text and showing 'Read More'—the result in the event list is fine.

![Image](https://github.com/user-attachments/assets/b8f349e1-e4a5-4684-b5cf-bb11afbfff1a)

Do you want to have a alternative text if introtext is empty?

**Comment by mckillo** [comment]
In the JEM code, there is a event_alternative_more, but it was created, neither used.

![Image](https://github.com/user-attachments/assets/dd31ab07-5ea9-48b5-8477-c95ad16dd7aa)

And it was to apply to length of title.

**Comment by mckillo** [comment]
I think so; it's fine to be operative and this old issue can be closed.

**Comment by jojo12** [comment]
I think, not all enter an event via simplelist. And also there is not always only one event creator 
So guests who visit an event directly, can't always see the introtext (before the readmore).

What when the admin decides to accept an introtext in simplelist, then other creators must always use this.

**Comment by jojo12** [comment]
I tried: I can use the readmore button as firstline in event description. (then the whole rest of the text is saved in fulltext.
then there's no intratext in simplelist (even when others use it) and the whole fulltext is visible in eventdescription.

But I think, we can close now! It's a good additional feature (I am not sure, if me personally I will work with.)
